docs/agents/sampling.md

```diff
@@ -0,0 +1,64 @@
+# Sampling
+
+To reduce processing and storage overhead, transactions may be sampled by agents.
+Sampling here refers to "head-based sampling".
+Head-based sampling is where a sampling decision is made at the root of the distributed trace, before the details or outcome of the trace are known, and propagated throughout the trace.
+
+## Configuration
+
+Agents can be configured to sample probabilistically, by specifying a sampling probability in the range \[0,1\] using the configuration `ELASTIC_APM_TRANSACTION_SAMPLE_RATE`. e.g.
+
+ - `ELASTIC_APM_TRANSACTION_SAMPLE_RATE=1` means all transactions will be sampled (the default)
+ - `ELASTIC_APM_TRANSACTION_SAMPLE_RATE=0` means no transactions will be sampled
+ - `ELASTIC_APM_TRANSACTION_SAMPLE_RATE=0.5` means approximately 50% of transactions will be sampled
+
+## Sampling rate
+
+At the time of making a sampling decision, the sampling rate must be recorded so that it can be associated with every transaction and span in the trace.
+This will be used by the server for scaling transaction and span metrics.
+
+The sampling rate must be recorded on transactions and spans as `sample_rate`. e.g.
```

Comment:
I think this could help to connect the dots on why this is important and how it is used:

```suggestion
This will be used by the server for scaling transaction and span metrics.
These metrics are used in the service map to display performance and throughput both from the perspective of the service itself (transaction metrics grouped by `service.name`) and from the perspective of upstream services (exit span metrics grouped by `destination.service.resource`).

These metrics will be more accurate when the sampling rate is high.
But even with lower sampling rates,
the server is able to calculate representative metrics.
To enable the server to properly weigh the metrics, agents must record the sampling rate on transactions and spans as `sample_rate`. e.g.
```

Reply:
SGTM, thanks!